sms_no_automatic_delete/tests/test_purge_and_unlink_sms.py

```diff
@@ -5,30 +5,50 @@
 
 from odoo.tests.common import TransactionCase
 
+from odoo.addons.sms.tests.common import MockSMS
 
-class TestSmsPurge(TransactionCase):
+
+class TestSmsPurgeAndUnlink(TransactionCase, MockSMS):
     def setUp(self):
-        super(TestSmsPurge, self).setUp()
+        super(TestSmsPurgeAndUnlink, self).setUp()
         self.sms_sms = self.env["sms.sms"]
         self.sms1 = self.sms_sms.create(
             {
-                "state": "sent",
+                "state": "outgoing",
                 "write_date": datetime.now() - timedelta(days=30),
             }
         )
         self.sms2 = self.sms_sms.create(
             {
-                "state": "sent",
+                "state": "outgoing",
                 "write_date": datetime.now() - timedelta(days=61),
             }
         )
         self.sms3 = self.sms_sms.create(
             {
-                "state": "sent",
+                "state": "outgoing",
                 "write_date": datetime.now() - timedelta(days=91),
             }
         )
 
+    def test_sms_state_as_sent(self):
+        with self.mockSMSGateway():
+            self.sms1.send()
+        self.assertEqual(self.sms1.state, "sent")
+        self.assertEqual(self.sms2.state, "outgoing")
+
+        with self.mockSMSGateway():
+            self.sms2.send()
+        self.assertEqual(self.sms2.state, "sent")
+
+    def test_sms_unlink(self):
+        # force_unlink=False by default
+        self.sms1.unlink()
+        self.assertEqual(len(self.sms_sms.search([])), 3)
```

Comment:
do not do a search, as this test will have random result.
If a module create a sms and so you already have an sms the qty will be 4.

you can simply do

```
self.sms1.unlink()
self.assertTrue(self.sms1.exists())
self.sms1.with_context(force_unlink=True).unlink()
self.assertFalse(self.sms1.exists())
```

Reply:
Indeed, it's better your way, I did the modification